tests/test_windows.py

```diff
@@ -0,0 +1,175 @@
+import rasterio
+
+import numpy as np
+import pytest
+
+from rasterio.windows import (
+    window, window_bounds, window_transform, eval_window, window_index, window_shape)
+
+def test_window_method():
```

Comment:
@perrygeo I think this test function should focus on testing the functions defined in `rasterio.windows`; those that become methods via the mixin should probably be in a different test module focused on testing the `RasterReader` class.